compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -563,6 +563,15 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                         }
                     }
                 }
+            } else if err_code == &rustc_errors::error_code!(E0412) {
+                if let Some(correct) = Self::dectect_wrong_primitive(path) {
+                    err.span_suggestion(
+                        span,
+                        "try using a Rust type instead",
```

Comment:
I'm not sure the phrase "Rust type" is very helpful.
```suggestion
                        "perhaps you intended to use this type",
```